github4s/src/test/scala-2/github4s/unit/EncoderDecoderSpec.scala

```diff
@@ -0,0 +1,156 @@
+package github4s.unit
+
+import cats.syntax.all._
+import com.eed3si9n.expecty.Expecty
+import github4s.ArbitraryDerivation
+import github4s.Decoders._
+import github4s.Encoders._
+import github4s.domain._
+import io.circe.{Decoder, Encoder, Printer}
+import org.scalacheck.fortyseven.GenInstances._
+import org.scalacheck.{Arbitrary, Gen}
+import org.scalactic.source.Position
+import org.scalatest.Assertion
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
+
+import scala.reflect.ClassTag
+
+class EncoderDecoderSpec extends AnyFlatSpec with ScalaCheckPropertyChecks {
```

Comment:
Why is this test in the scala-2 directory? You've included `ArbitraryDerivation` for both Scala 2 or 3 - is there some other dependency that's missing for Scala 3?

Reply:
scalacheck-shapeless doesn't exist for scala3, since on scala3, shapeless isn't out.

I put the spec in scala2 because I ran into what I think might be a compiler bug, preventing me from using the (otherwise working) scala-3 `ArbitraryDerivation`.

https://github.com/lampepfl/dotty/issues/13001